### PR TITLE
Support torch.nn.Module

### PR DESCRIPTION
**Pull request by xutongye**
## Description
Support serde and send of `torch.nn.Module` objects, including pre-defined `Module` types in `torch` like `Linear/Sequential/...` and user defined models subclassed from `nn.Module`.

PS: This PR is directely based on #5230 , because I am considering taking use of `Plan` to support remote call of user defined models.

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- examples/experimental/tongye/torch_nn_Module.ipynb
## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by koenvanderveen** [comment]
I wonder whether we really need to serialize the forward method as a plan here. Alternatively, we could just serialize the state of the module, call the .forward method on the module pointer, and trace the actions that originate from that call. So a module in a plan instead of a plan in a module.

**Review by madhavajay — APPROVED** [review]
I have added the notebook examples to tests and I think this is good to merge. We should discuss how best to use this in combination with Plans and the existing Sy.Module.
